components/domains/autorenewal.tsx

```diff
@@ -0,0 +1,318 @@
+import Button from "../UI/button";
+import {
+  useAccount,
+  useContractRead,
+  useContractWrite,
+} from "@starknet-react/core";
+import {
+  formatHexString,
+  isValidEmail,
+  selectedDomainsToArray,
+} from "../../utils/stringService";
+import { applyRateToBigInt } from "../../utils/feltService";
+import { Abi, Call } from "starknet";
+import { posthog } from "posthog-js";
+import styles from "../../styles/components/registerV2.module.css";
+import TextField from "../UI/textField";
+import SwissForm from "./swissForm";
+import Wallets from "../UI/wallets";
+import {
+  computeMetadataHash,
+  generateSalts,
+} from "../../utils/userDataService";
+import {
+  areDomainSelected,
+  getPriceFromDomains,
+  getPriceFromDomain,
+} from "../../utils/priceService";
+import AutoRenewalDomainsBox from "./autoRenewalDomainsBox";
+import autoRenewalCalls from "../../utils/callData/autoRenewalCalls";
+import BackButton from "../UI/backButton";
+import { useRouter } from "next/router";
+import { useNotificationManager } from "../../hooks/useNotificationManager";
+import {
+  NotificationType,
+  TransactionType,
+  UINT_128_MAX,
+  swissVatRate,
+} from "../../utils/constants";
+import RegisterCheckboxes from "../domains/registerCheckboxes";
+import { utils } from "starknetid.js";
+import { useEtherContract } from "../../hooks/contracts";
+import RegisterConfirmationModal from "../UI/registerConfirmationModal";
+
+type SubscriptionProps = {
+  groups: string[];
+};
+
+const Subscription: FunctionComponent<SubscriptionProps> = ({ groups }) => {
+  const [email, setEmail] = useState<string>("");
+  const [emailError, setEmailError] = useState<boolean>(true);
+  const [isSwissResident, setIsSwissResident] = useState<boolean>(false);
+  const [salesTaxRate, setSalesTaxRate] = useState<number>(0);
+  const [salesTaxAmount, setSalesTaxAmount] = useState<string>("0");
+  const [callData, setCallData] = useState<Call[]>([]);
+  const [price, setPrice] = useState<string>("");
+  const [isTxModalOpen, setIsTxModalOpen] = useState(false);
+  const [termsBox, setTermsBox] = useState<boolean>(true);
+  const [renewalBox, setRenewalBox] = useState<boolean>(true);
+  const [walletModalOpen, setWalletModalOpen] = useState<boolean>(false);
+  const [salts, setSalts] = useState<string[] | undefined>();
+  const [metadataHashes, setMetadataHashes] = useState<string[] | undefined>();
+  const [selectedDomains, setSelectedDomains] =
+    useState<Record<string, boolean>>();
+  const { address } = useAccount();
+  const { writeAsync: execute, data: autorenewData } = useContractWrite({
+    calls: callData,
+  });
+  const [domainsMinting, setDomainsMinting] =
+    useState<Record<string, boolean>>();
+  const { addTransaction } = useNotificationManager();
+  const router = useRouter();
+  const duration = 1;
+  const { contract: etherContract } = useEtherContract();
+  const { data: erc20AllowanceData, error: erc20AllowanceError } =
```

Comment:
Can we make something in utils to manage this allowance check ? It seems we're always using the same code :)